### PR TITLE
Issue #KN-231 feat: Added the support for google-cloud.

### PR DESCRIPTION
**Pull request by AmiableAnil**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/KN-231" title="KN-231" target="_blank">RFCKN-231</a>  Integrate cloud-store-sdk for google cloud
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Java 11, scala-2.11, play-2.7.2
* Hardware versions: 2 CPU/ 4GB RAM

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: content-api/content-actors/src/main/scala/org/sunbird/content/actors/ContentActor.scala [hunk 1]
```diff
@@ -144,6 +144,7 @@ class ContentActor @Inject() (implicit oec: OntologyEngineContext, ss: StorageSe
 	def uploadPreSignedUrl(request: Request): Future[Response] = {
 		val `type`: String = request.get("type").asInstanceOf[String].toLowerCase()
 		val fileName: String = request.get("fileName").asInstanceOf[String]
+		val mimeType: String = request.get("mimeType").asInstanceOf[String]
```
Comment (vinukumar-vs) [inline content-api/content-actors/src/main/scala/org/sunbird/content/actors/ContentActor.scala:147]: Why are we looking for mimeType here?

File: content-api/content-actors/src/main/scala/org/sunbird/content/actors/ContentActor.scala [hunk 2]
```diff
@@ -152,7 +153,7 @@ class ContentActor @Inject() (implicit oec: OntologyEngineContext, ss: StorageSe
 			val objectKey = if (StringUtils.isEmpty(filePath)) "content" + File.separator + `type` + File.separator + identifier + File.separator + Slug.makeSlug(fileName, true)
 				else filePath + File.separator + "content" + File.separator + `type` + File.separator + identifier + File.separator + Slug.makeSlug(fileName, true)
 			val expiry = Platform.config.getString("cloud_storage.upload.url.ttl")
-			val preSignedURL = ss.getSignedURL(objectKey, Option.apply(expiry.toInt), Option.apply("w"))
+			val preSignedURL = ss.getSignedURL(objectKey, Option.apply(expiry.toInt), Option.apply("w"), Option.apply(mimeType))
```
Comment (vinukumar-vs) [inline content-api/content-actors/src/main/scala/org/sunbird/content/actors/ContentActor.scala:156]: Why are we sending mimetype information to the library? Liberary should not check this property..

@maheshkumargangula

File: platform-modules/mimetype-manager/src/main/scala/org/sunbird/cloudstore/StorageService.scala [hunk 3]
```diff
@@ -44,17 +43,13 @@ class StorageService {
...
-        else if (StringUtils.equalsIgnoreCase(storageType, "aws"))
-            Platform.config.getString("aws_storage_container")
-        else if (StringUtils.equalsIgnoreCase(storageType, "gcloud"))
-          Platform.config.getString("gcloud_storage_bucket")
-        else if (StringUtils.equalsIgnoreCase(storageType, "cephs3"))
-            Platform.config.getString("cephs3_storage_container")
-        else
-            throw new ServerException("ERR_INVALID_CLOUD_STORAGE", "Container name not configured.")
+    def getContainerName: String = {
+      storageType match {
```
Comment (vinukumar-vs) [inline platform-modules/mimetype-manager/src/main/scala/org/sunbird/cloudstore/StorageService.scala:47]: Good to see code improvements

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

**Review by vinukumar-vs — APPROVED** [review]

File: content-api/content-actors/src/main/scala/org/sunbird/content/actors/ContentActor.scala (hunk 1 above)
Comment (vinukumar-vs) [reply content-api/content-actors/src/main/scala/org/sunbird/content/actors/ContentActor.scala:147]: Only for GCP we are sending. For AWS & Azure even though request doesn't have, it's not breaking.

**Review by vinukumar-vs — APPROVED** [review]

**Comment by AmiableAnil** [comment]
@gandham-santhosh We have added the GCP related changes to this PR and also renamed the few method. For some of the class unit test was not available and I have added the unit test for the recent changes. Currently coverage is 46.7% for this PR. I request you to please merge this PR. We will add the coverage in the upcoming release.

**Pull request merged**